homeassistant/util/color.py

```diff
@@ -90,3 +94,73 @@ def rgb_hex_to_rgb_list(hex_string):
             for i in range(0,
                            len(hex_string),
                            len(hex_string) // 3)]
+
+
+def color_temperature_to_rgb(color_temperature_kelvin):
+    """
+    Return an RGB color from a color temperature in Kelvin.
+
+    This is a rough approximation based on the formula provided by T. Helland
+    http://www.tannerhelland.com/4435/convert-temperature-rgb-algorithm-code/
+    """
+    # range check
+    if color_temperature_kelvin < 1000:
+        color_temperature_kelvin = 1000
+    elif color_temperature_kelvin > 40000:
+        color_temperature_kelvin = 40000
+
+    tmp_internal = color_temperature_kelvin / 100.0
+
+    red = _get_red(tmp_internal)
+
+    green = _get_green(tmp_internal)
+
+    blue = _get_blue(tmp_internal)
+
+    return (red, green, blue)
+
+
+def _bound(color_component, minimum=0, maximum=255):
```

Comment:
I'm not sure what the usefulness is of making this an extra method. Why not just inline the logic